### PR TITLE
Snom plugins update

### PR DESCRIPTION
**Pull request by ymolinet**
Add Support of D717 and D385 (tested)
Add firmware 10.1.46.16 (tested on D385, D735 and D785), 10.1.49.11, 10.1.51.11 (tested on D385, D735 and D785) and 10.1.54.13 (tested on D385, D735 and D785)
Add support of UXM firmware update

### CODE REVIEW

**Comment by ymolinet** [comment]
I'm using the firmware 10.1.46.16 in production on many Wazo 18.03 and Wazo Plateform 20.07 (D385, D735)

**Comment by ymolinet** [comment]
Thanks for your feedback. I fix quickly all this point this week.

File: plugins/wazo-snom/tools/generate_pkgs-db.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+if [ "$#" -ne 1 ]; then
+    echo "specify firmware version"
+    exit
+fi
+
+dir=/home/dixinfor/snom-fw/$1
+rm $dir -fr
+mkdir -p $dir
+cd $dir
+touch $dest_filename
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/tools/generate_pkgs-db.sh:18]: Fix in my commit 6f3d428

File: plugins/wazo-snom/tools/generate_pkgs-db.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+dest_filename="pkgs.db"
+echo $dest_filename
+
+if [ "$#" -ne 1 ]; then
+    echo "specify firmware version"
+    exit
+fi
+
+dir=/home/dixinfor/snom-fw/$1
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/tools/generate_pkgs-db.sh]: Fix in my commit 6f3d428

File: plugins/wazo-snom/tools/generate_pkgs-db.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+for i in "${snom_products[@]}" 
+do
+ 	NAME=$i
+        if [[ ${i:0:1} == "D" ]]; then
+	        NAME="${i:1}"
+        fi
+	echo "[pkg_$NAME-fw]" >> $dest_filename
+        echo "description: Firmware for Snom $i" >> $dest_filename
+        echo "description_fr: Firmware pour Snom $i" >> $dest_filename
+	echo "version: $1" >> $dest_filename
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/tools/generate_pkgs-db.sh:29]: Fix in my commit 6f3d428

File: plugins/wazo-snom/10.1.39.11/entry.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 The Wazo Authors  (see AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0+
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/10.1.39.11/entry.py]: Fix in 167abdd

File: plugins/wazo-snom/10.1.39.11/entry.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 The Wazo Authors  (see AUTHORS file)
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/10.1.39.11/entry.py]: fix in 167abdd

File: plugins/wazo-snom/build.py [hunk 6]
```diff
@@ -267,3 +267,264 @@ def build_10_1_26_1(path):
...
+                       stdout=f)
+
+    check_call(['rsync', '-rlp', '--exclude', '.*',
+                '10.1.46.16/', path])
+
+				
+@target('10.1.49.11', 'wazo-snom-10.1.49.11')
+def build_10_1_49_11(path):
+    MODELS = [
+	    ('D120', 'r'),
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/build.py]: fix in 6386759

File: plugins/wazo-snom/10.1.46.16/entry.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+# Copyright 2018 The Wazo Authors  (see AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0+
+
+common_globals = {}
+execfile_('common.py', common_globals)
+
+MODELS = [
+	u'D375',
+    u'D385',
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/10.1.46.16/entry.py]: fix in 2583efb

File: plugins/wazo-snom/10.1.46.16/plugin-info [hunk 8]
```diff
@@ -0,0 +1,135 @@
+{
+    "version": "0.3.0",
+    "description": "Plugin for Snom D3x5, D717, D7x5 in version 10.1.46.16",
+    "description_fr": "Greffon pour Snom Snom D3x5, D717, D7x5 en version 10.1.46.16",
+    "capabilities": {
+        "Snom, D120, 10.1.46.16": {
+            "lines": 12,
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/10.1.46.16/plugin-info]: fix in a7ff0c1

File: plugins/wazo-snom/10.1.46.16/plugin-info [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+			"high_availability": false,
+            "function_keys": 8,
+            "expansion_modules": 0,
+            "switchboard": false,
+            "protocol": "sip"
+        },
+		"Snom, D345, 10.1.46.16": {
+            "lines": 12,
+			"high_availability": false,
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/10.1.46.16/plugin-info]: fix in a7ff0c1

File: plugins/wazo-snom/10.1.39.11/plugin-info [hunk 10]
```diff
@@ -0,0 +1,31 @@
+{
+    "version": "0.2.0",
+    "description": "Plugin for Snom D3x5, D717, D7x5 in version 10.1.39.11.",
+    "description_fr": "Greffon pour Snom Snom D3x5, D717, D7x5 en version 10.1.39.11.",
+    "capabilities": {
+        "Snom, D375, 10.1.39.11": {
+            "sip.lines": 12
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/10.1.39.11/plugin-info]: fix in a7ff0c1

File: plugins/wazo-snom/common/templates/common/snom-model-firmware.xml.tpl.btpl [hunk 11]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<firmware-settings>
+<<<<<<< HEAD:plugins/wazo-snom/common/templates/common/snom-model-firmware.xml.tpl.btpl
+  <firmware>http://{{ ip }}:{{ http_port }}/firmware/#FW_FILENAME#</firmware>
+  <firmware_uxm>http://{{ ip }}:{{ http_port }}/firmware/snomUXM-2.1.1.bin</firmware_uxm>
+=======
```
[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/common/templates/common/snom-model-firmware.xml.tpl.btpl:6]: I'm not sure to understand what do you mean...

[the remark this answers is not in the document]
Comment (bloom1) [reply plugins/wazo-snom/common/templates/common/snom-model-firmware.xml.tpl.btpl:6]: The
```
<<<<<<< HEAD:filename
  CODE
=======
  CODE
>>>>>>> XXXX:filename
```
structure is a diff inserted by git to tell you there was a conflict after an operation (merge, rebase, stash pop, etc.) and that you have to choose between the two.

[the remark this answers is not in the document]
Comment (ymolinet) [reply plugins/wazo-snom/common/templates/common/snom-model-firmware.xml.tpl.btpl]: Fix in c21b090

**Comment by ymolinet** [comment]
up ?

File: plugins/xivo-cisco-sccp/legacy/var/tftpboot/RINGLIST.XML [hunk 12]
```diff
@@ -1,2 +1 @@
-<CiscoIPPhoneRingList>
-</CiscoIPPhoneRingList>
+RINGLIST.XML
```
Comment (bloom1) [inline plugins/xivo-cisco-sccp/legacy/var/tftpboot/RINGLIST.XML:1]: I just noticed this: could you do a separate PR for this and explain why? It's not in the same plugin. Thank you!

Comment (ymolinet) [reply plugins/xivo-cisco-sccp/legacy/var/tftpboot/RINGLIST.XML:1]: It's an issue, you can remove it....

**Comment by ymolinet** [comment]
Thanks a lot !
